src/Forms/GridField/GridFieldStateManager.php

```diff
@@ -41,12 +42,14 @@ public function addStateToURL(GridField $gridField, string $url): string
         $key = $this->getStateKey($gridField);
         $value = $gridField->getState(false)->Value();
 
+        $hash = GridFieldSessionState::singleton()->setValueByHash($key, $value,  "md5");
+
         // Using a JSON-encoded empty array as the blank value, to avoid changing Value() semantics in a minor release
         if ($value === '{}') {
             return $url;
         }
 
-        return HTTP::setGetVar($key, $value, $url);
+        return HTTP::setGetVar($key, $hash, $url);
```

Comment:
This is a breaking change. We can't do this, as nice as it would be. If someone's relying on the current JSON values being in the URL for their custom functionality, changing this to a hash will break it.

Looking at URL character limits for various CDNs (which will be lower than browsers typically allow) I don't think it will be a problem:
- [Amazon CloudFront allows 8kb](https://docs.aws.amazon.com/AmazonCloudFront/latest/DeveloperGuide/cloudfront-limits.html)
- [Imperva allows 4kb](https://docs.imperva.com/bundle/on-premises-knowledgebase-reference-guide/page/extremely_long_url_parameter.htm)

So taking the lowest of these, 4kb gives us 4000 characters in the URL for ascii characters or, in the worst case, 1000 characters for some UTF-8 characters. I think we will be fine with that.

We can always look at implementing the hash functionality in CMS5 as a separate action, since we're allowed to break things for a major version.

Reply:
I've already removed Hash logic and just keeping State as it is in the URL.